lib/charms/data_platform_libs/v0/upgrade.py

```diff
@@ -606,6 +606,21 @@ def upgrade_stack(self, stack: List[int]) -> None:
         self.peer_relation.data[self.charm.app].update({"upgrade-stack": json.dumps(stack)})
         self._upgrade_stack = stack
 
+    @property
+    def other_unit_states(self) -> list:
+        """Current upgrade state for other units.
```

Comment:
these are a probable fix for an issue, still to be confirmed. 
If so, we will have a PR on upgrade libs before this get merged